resource_detectors/lib/opentelemetry/resource/detectors/azure.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+require 'net/http'
+
+module OpenTelemetry
+  module Resource
+    module Detectors
+      # Azure contains detect class method for determining Azure environment resource attributes
+      module Azure
+        extend self
+
+        AZURE_METADATA_URI = 'http://169.254.169.254/metadata/instance/compute?api-version=2019-08-15'
```

Comment:
Thank you for the link to the docs! I am not familiar with the Azure API so that was very helpful. 

Thank you for the suggestion for implementing #20 as well!